src/test/java/org/jenkinsci/plugins/matrixauth/PermissionAdderTest.java

```diff
@@ -43,10 +42,8 @@ public void ensureSavingAfterInitialUser() throws Throwable {
                 signup.submit(j);
                 User alice = User.get("alice", false, Collections.emptyMap());
                 Assert.assertNotNull(alice);
-                Assert.assertTrue(j.jenkins.getACL().hasPermission(alice.impersonate(), Jenkins.ADMINISTER));
-        });
-        sessions.then(j -> {
-                Assert.assertTrue(j.jenkins.getACL().hasPermission(User.get("alice", false, Collections.emptyMap()).impersonate(), Jenkins.ADMINISTER));
+                Assert.assertTrue(j.jenkins.getACL().hasPermission2(alice.impersonate2(), Jenkins.ADMINISTER));
         });
+        sessions.then(j -> Assert.assertTrue(j.jenkins.getACL().hasPermission2(User.getOrCreateByIdOrFullName("alice").impersonate2(), Jenkins.ADMINISTER)));
```

Comment:
Avoid `getOrCreateByIdOrFullName` unless you truly intend this.
```suggestion
        sessions.then(j -> Assert.assertTrue(j.jenkins.getACL().hasPermission2(User.getById("alice", true).impersonate2(), Jenkins.ADMINISTER)));
```